test/blocklib/core/fft/fft.hpp

```diff
@@ -179,160 +89,105 @@ struct fft : node<fft<T>> {
+        std::transform(outData.begin(), outData.end(), std::next(ds.signal_values.begin(), N), [](const auto &c) { return c.imag(); });
+        std::copy_n(magnitudeSpectrum.begin(), N, std::next(ds.signal_values.begin(), 2U * N));
+        std::copy_n(phaseSpectrum.begin(), N, std::next(ds.signal_values.begin(), 3U * N));
+
+        ds.signal_ranges.resize(4);
+        for (std::size_t i = 0; i < 4; i++) {
+            const auto mm = std::minmax_element(std::next(ds.signal_values.begin(), static_cast<std::ptrdiff_t>(i * N)), std::next(ds.signal_values.begin(), static_cast<std::ptrdiff_t>((i + 1U) * N)));
+            ds.signal_ranges[i] = { *mm.first, *mm.second };
         }
 
         ds.signal_errors    = {};
-        ds.meta_information = {
-            { { "fftSize", fftSize }, { "window", window }, { "outputInDb", outputInDb }, { "numerator", this->numerator }, { "denominator", this->denominator }, { "stride", this->stride } }
-        };
+        ds.meta_information = { { { "sampleRate", sampleRate },
+                                  { "fftSize", fftSize },
+                                  { "window", window },
+                                  { "outputInDb", outputInDb },
+                                  { "numerator", this->numerator },
+                                  { "denominator", this->denominator },
+                                  { "stride", this->stride } } };
 
         return ds;
     }
 
     void
     prepareInput() {
-        static_assert(sizeof(InDataType) == sizeof(InHistoryType), "std::complex<T> and T[2] must have the same size");
-        std::memcpy(fftwIn.get(), &(*inputHistory.begin()), sizeof(InDataType) * fftSize);
+        std::copy_n(inputHistory.begin(), fftSize, inData.begin());
         // apply window function if needed
         if (window != static_cast<int>(WindowFunction::None)) {
-            if (fftSize != windowVector.size()) throw std::runtime_error(fmt::format("fftSize({}) and windowVector.size({}) are not equal.", fftSize, windowVector.size()));
+            if (fftSize != windowVector.size()) {
+                throw std::runtime_error(fmt::format("fftSize({}) and windowVector.size({}) are not equal.", fftSize, windowVector.size()));
+            }
             for (std::size_t i = 0; i < fftSize; i++) {
                 if constexpr (ComplexType<T>) {
-                    fftwIn[i][0] *= windowVector[i];
-                    fftwIn[i][1] *= windowVector[i];
+                    inData[i].real(inData[i].real() * windowVector[i]);
+                    inData[i].imag(inData[i].imag() * windowVector[i]);
                 } else {
-                    fftwIn[i] *= windowVector[i];
+                    inData[i] *= windowVector[i];
                 }
             }
         }
     }
 
     void
-    computeFft() {
-        fftw<T>::execute(fftwPlan.get());
+    computeFFT() {
```

Comment:
let's inline this single line function... keeps the code more local and easier to reason about.